todo_app/data/trello_items.py

```diff
@@ -0,0 +1,146 @@
+
+    if response.status_code != 200:
+        raise ValueError(f'Failed to create todo item for "{title}". Received status code {response.status_code}')
+    
+    response_body = response.json()
+    return Item(id=response_body['id'], title=title, status=ItemStatus.NOT_STARTED)
+
+def remove_item(id: str):
+    response = requests.delete(f'{base_uri}/cards/{id}', params=query_params)
+    if response.status_code != 200:
+        raise ValueError(f'Failed to delete todo item with id "{id}"". Received status code {response.status_code}')
+
+def update_item(item: Item):
+    lists = get_lists()
+    to_do_list_id = get_list_id_or_throw(lists, "To Do")
+    doing_list_id = get_list_id_or_throw(lists, "Doing")
+    done_list_id = get_list_id_or_throw(lists, "Done")
+    
+    list_id_to_move_to = None
+    if item.status == ItemStatus.NOT_STARTED: list_id_to_move_to = to_do_list_id
+    elif item.status == ItemStatus.IN_PROGRESS: list_id_to_move_to = doing_list_id
+    elif item.status == ItemStatus.COMPLETE: list_id_to_move_to = done_list_id
+
+    response = requests.put(f'{base_uri}/cards/{item.id}', params={**query_params, 'idList': list_id_to_move_to})#
+
+    if response.status_code != 200:
+        raise ValueError(f'Failed to update todo item with id "{item.id}". Received status code {response.status_code}')
+
+
+def get_items() -> List[Item]:
+    """
+    Fetches all saved items from trello.
+
+    Returns:
+        list: The list of saved items.
+    """
+    lists = get_lists()
+    to_do_list_id = get_list_id_or_throw(lists, "To Do")
+    doing_list_id = get_list_id_or_throw(lists, "Doing")
+    done_list_id = get_list_id_or_throw(lists, "Done")
+
+    to_do_items = get_items_from_list(to_do_list_id, 'Not Started')
+    in_progress_items = get_items_from_list(doing_list_id, 'In Progress')
+    complete_items = get_items_from_list(done_list_id, 'Complete')
+
+    return to_do_items + in_progress_items + complete_items
+
+def get_item(id: str) -> Optional[Item]:
+    """
+    Fetches the saved item with the specified ID.
+
+    Args:
+        id: The ID of the item.
+
+    Returns:
+        item: The saved item, or None if no items match the specified ID.
+    """
+    items = get_items()
+    return next((item for item in items if item.id == id), None)
+
+
+def add_item(title: str) -> Item:
+    """
+    Adds a new item with the specified title to the todo list.
+
+    Args:
+        title: The title of the item.
+
+    Returns:
+        item: The saved item.
+    """
+
+    # Add the item to the list
+    item = create_todo_item(title)
+
+    return item
+
+
+def delete_item(id: str) -> Item:
+    """ 
+    Deletes the saved item with the specified ID.
+
+    Args:
+        id: The ID of the item.
+
+    Returns:
+        item: The saved item, or None if no items match the specified ID.
+    """
+    item = get_item(id)
+    remove_item(id)
+    return item
```

Comment:
I suspect this follows the existing pattern, but worth noting that no existing code actually uses the fact we return the item here, so that call to Trello could easily be cut out